pages/desktop/developer_hub/content_tools.py

```diff
@@ -0,0 +1,107 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from pages.desktop.developer_hub.base import Base
+from pages.page import Page, PageRegion
+
+
+class ContentTools(Base):
+
+    _submit_new_add_on_button_locator = (By.CSS_SELECTOR, '.landing--submit a')
+    _agree_button_locator = (By.CSS_SELECTOR, '.sign button')
+    _submit_add_on_form_button_locator = (By.CSS_SELECTOR, '.addon-upload button[type="submit"]')
+    _select_add_on_file_button_locator = (By.CLASS_NAME, 'form-inline--file-input')
+    _add_on_file_input_locator = (By.ID, 'submission-addon--zip')
+    _notification_message_locator = (By.CSS_SELECTOR, '.notification')
+    _add_on_locator = (By.CSS_SELECTOR, 'li.addon-for-listing')
+
+    @property
+    def header(self):
+        return self.HeaderRegion(self.testsetup)
+
+    @property
+    def notification_message(self):
+        return self.selenium.find_element(*self._notification_message_locator).text
+
+    @property
+    def add_ons(self):
+        return [self.AddOn(self.testsetup, web_element)
+                for web_element in self.selenium.find_elements(*self._add_on_locator)]
+
+    def add_on(self, name):
+        return next(a for a in self.add_ons if a.name == name)
+
+    def go_to_page(self):
+        self.selenium.get(self.base_url + '/content')
+        return self
+
+    def click_submit_new_add_on(self):
+        self.selenium.find_element(*self._submit_new_add_on_button_locator).click()
+        el = self.selenium.find_element(*self._agree_button_locator)
+        WebDriverWait(self.selenium, self.timeout).until(EC.visibility_of(el))
+
+    def click_agree(self):
+        self.selenium.find_element(*self._agree_button_locator).click()
+        el = self.selenium.find_element(*self._select_add_on_file_button_locator)
+        WebDriverWait(self.selenium, self.timeout).until(EC.visibility_of(el))
+
+    def select_add_on_file(self, zip_file):
+        file_input = self.selenium.find_element(*self._add_on_file_input_locator)
+        # WARNING: This is a huge hack but is required in order to be able to send_keys
+        # to the file input, which has an explicit top of -9999px
+        self.selenium.execute_script('arguments[0].style.top = 0;', file_input)
```

Comment:
Hi,

I just happened to notice this from SeleniumHQ/selenium#706.

I'd also suggest to re-set the value of the `top` property to the original one right after sendKeys: otherwise, it could overlap with another element you might want to interact with afterwards.

```
initialTopValue = self.selenium.get_css_value("top")
self.selenium.execute_script('arguments[0].style.top = 0;', file_input)
file_input.send_keys(zip_file)
self.selenium.execute_script('arguments[0].style.top = ' + initialTopValue + ';', file_input)
```